python/pyspark/pandas/internal.py

```diff
@@ -962,9 +1025,9 @@ def field_for(self, label: Label) -> InternalField:
             raise KeyError(name_like_string(label))
 
     @property
-    def spark_frame(self) -> SparkDataFrame:
+    def spark_frame(self) -> LegacyDataFrame:
```

Comment:
`GenericDataFrame`?

Reply:
Maybe can I address remaining mypy check including this in the follow-up, because updating this brings another hundreds of mypy errors?
Btw, there are too many mypy errors to handle all typing at once in this PR, so I created a separate ticket to address remaining typing within subsequent tasks: SPARK-43045